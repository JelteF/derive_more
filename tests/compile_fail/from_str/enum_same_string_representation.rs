#![allow(non_camel_case_types)]

#[derive(derive_more::FromStr)]
enum Enum {
    abc,
    #[from_str(rename_all = "lowercase")]
    AB_C,
}

#[derive(derive_more::FromStr)]
#[from_str(rename_all = "SCREAMING_SNAKE_CASE")]
enum EnumTop {
    BaR,
    BA_R,
}

fn main() {}
