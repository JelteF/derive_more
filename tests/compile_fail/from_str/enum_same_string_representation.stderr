error: `abc`, `AB_C` variants cannot have the same "abc" string representation
 --> tests/compile_fail/from_str/enum_same_string_representation.rs:4:6
  |
4 | enum Enum {
  |      ^^^^

error: `BaR`, `BA_R` variants cannot have the same "BA_R" string representation
  --> tests/compile_fail/from_str/enum_same_string_representation.rs:12:6
   |
12 | enum EnumTop {
   |      ^^^^^^^
